apps/settings/js/panels/operator_settings/models/mobile_connection_wrapper.js

```diff
@@ -0,0 +1,127 @@
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+
+  // Mobile connection state
+  const STATE = {
+    IDLE: 0,
+    BUSY: 1
+  };
+
+  // A promise that never resolves.
+  const PENDING_PROMISE = new Promise(() => {});
+
+  function Task(promise) {
+    var canceled = false;
+    var task = new Promise((resolve, reject) => {
+      promise.then((result) => {
+        if (!canceled) {
+          resolve(result);
+        } else {
+          reject('canceled');
```

Comment:
I think `canceled` can be extracted as a property which resides inside STATE or we can introduce a new constant for it.